src/bootstrap/src/core/config/config.rs

```diff
@@ -744,27 +745,48 @@ enum ReplaceOpt {
 }
 
 trait Merge {
-    fn merge(&mut self, other: Self, replace: ReplaceOpt);
+    fn merge(&mut self, extension_path: Option<PathBuf>, other: Self, replace: ReplaceOpt);
 }
 
 impl Merge for TomlConfig {
     fn merge(
         &mut self,
-        TomlConfig { build, install, llvm, gcc, rust, dist, target, profile, change_id }: Self,
+        extension_path: Option<PathBuf>,
+        TomlConfig { build, install, llvm, gcc, rust, dist, target, profile, change_id, include }: Self,
         replace: ReplaceOpt,
     ) {
         fn do_merge<T: Merge>(x: &mut Option<T>, y: Option<T>, replace: ReplaceOpt) {
             if let Some(new) = y {
                 if let Some(original) = x {
-                    original.merge(new, replace);
+                    original.merge(None, new, replace);
                 } else {
                     *x = Some(new);
                 }
             }
         }
 
-        self.change_id.inner.merge(change_id.inner, replace);
-        self.profile.merge(profile, replace);
+        for include_path in include.clone().unwrap_or_default() {
+            let included_toml = Config::get_toml(&include_path).unwrap_or_else(|e| {
+                eprintln!(
+                    "ERROR: Failed to parse default config profile at '{}': {e}",
+                    include_path.display()
+                );
+                exit!(2);
+            });
+
+            if let Some(extension_path) = &extension_path {
+                assert!(
+                    !included_toml.include.clone().unwrap_or_default().contains(extension_path),
+                    "Cyclic inclusion detected in extension: {}",
+                    extension_path.display()
+                );
+            }
```

Comment:
This will only work when cyclic inclusion happens between two direct extensions. I will rewrite this part later today or early tomorrow.